_posts/2019-02-04-js-calculator.html

```diff
@@ -0,0 +1,59 @@
+---
+title: Vanilla js calculator
+thumbnail: -SITE_URL-/images/calculator.jpg
+excerpt: "A copy of the windows calculator functionality. It is meant to be an exercise in vanilla javascript."
+# githublink: "#"
+bitbucketlink: "https://bitbucket.org/Dries_Sercu/vanilla-js-calculator"
+demolink: "-SITE_URL-/projects/vanilla-js-calculator/index.html"
+---
+<p>
+	It had been a while since I made anything using javascript. As an exercise I reproduced as much of the windows calculator functionality as possible.
+</p>
+<p>
+	I decided against using a framework because I wanted to focus on javascript instead of some framework. Using a framework might have made this easier but the focus of the exercise would have been different.
+</p>
+<p>
+	While I was at it I also decided to write tests and to generate reports about my tests.
+</p>
+<p>
+	The calculator has some nice features, and it's been a nice study in user experience.
+</p>
+
+<h2>Jasmine</h2>
+
+<p>
+	From previous experience i learned that, without realizing it, you will break your own application.
+	To make sure I could break pieces without knowing about it something was required to verify that every piece of code worked as expected.
```

Comment:
To make sure I could break pieces without knowing about it, something was required to verify that every piece of code worked as expected.

--> not sure if you meant what you said? "To make sure I could break pieces without knowing about it,"?